src/gui/attributetable/qgsdualview.cpp

```diff
@@ -348,6 +348,17 @@ int QgsDualView::filteredFeatureCount()
 
 void QgsDualView::viewWillShowContextMenu( QMenu* menu, const QModelIndex& atIndex )
 {
+  if ( !menu )
+  {
+    return;
+  }
+
+  QgsVectorLayer* vl = mFilterModel->layer();
+  if ( vl && vl->geometryType() != QGis::NoGeometry )
```

Comment:
I still think a safer approach would be to check if the current feature has a non-empty geometry here. It's possible the layer has a geometry type, but the active feature has no geometry.